### PR TITLE
Add description field to Nexus endpoint spec

### PR DESCRIPTION
**Pull request by nikki-dag**
https://temporalio.atlassian.net/browse/CLD-2996

### CODE REVIEW

File: app/nexus.go [hunk 1]
```diff
@@ -97,11 +99,15 @@ func (c *NexusClient) createEndpoint(
 
 func (c *NexusClient) patchEndpoint(
 	existingEndpoint *nexus.Endpoint,
+	description string,
 	targetNamespaceID string,
 	targetTaskQueue string,
 	resourceVersion string,
 	asyncOperationId string,
 ) (*operation.AsyncOperation, error) {
+	if description != "" {
```
Comment (mastermanu) [inline app/nexus.go:108]: what's the tcld command to clear an existing description?

File: app/nexus.go [hunk 2]
```diff
@@ -352,18 +366,20 @@ func NewNexusCommand(getNexusClientFn GetNexusClientFn) (CommandOut, error) {
 								"Only the fields that are provided will be updated",
 							Flags: []cli.Flag{
 								endpointNameFlag,
+								endpointDescriptionOptionalFlag,
```
Comment (mastermanu) [inline app/nexus.go:369]: do we need to update description field for command?

File: app/nexus.go [hunk 3]
```diff
@@ -323,6 +335,7 @@ func NewNexusCommand(getNexusClientFn GetNexusClientFn) (CommandOut, error) {
 								"This will fail if an endpoint with the same name is already registered",
 							Flags: []cli.Flag{
 								endpointNameFlag,
+								endpointDescriptionOptionalFlag,
```
Comment (mastermanu) [inline app/nexus.go:338]: do we need to update description field for command?

Comment (nikki-dag) [reply app/nexus.go:358]: Updated description.

File: app/nexus.go (hunk 2 above)
Comment (nikki-dag) [reply app/nexus.go:390]: Updated description.

File: app/nexus.go (hunk 1 above)
Comment (nikki-dag) [reply app/nexus.go:108]: Added ` nexus endpoint description set/unset` commands.
Now we have 2 ways of updating the description.
- Wanted to keep it in the `update` command to be consistent with `create`
- Wanted to keep the `set` to be consistent with `unset`

Happy to change if you think one of the ways is unnecessary.

Comment (nikki-dag) [reply app/nexus.go]: Changed this too `unset-description` bool flag similar to os cli.
https://github.com/bergundy/cli/blob/187b82da3e3d8702df476bc418abf9b706f7f95a/temporalcli/commands.gen.go#L979C5-L979C56

File: app/nexus.go (hunk 2 above)
Comment (nikki-dag) [reply app/nexus.go:423]: OS cli doesn't have an additional comment for the description field.
https://github.com/bergundy/cli/blob/187b82da3e3d8702df476bc418abf9b706f7f95a/temporalcli/commands.gen.go#L852

I think we can leave it for now.

File: app/nexus.go [hunk 4]
```diff
@@ -267,6 +304,28 @@ func NewNexusCommand(getNexusClientFn GetNexusClientFn) (CommandOut, error) {
 		Usage:    "Namespace that is allowed to call this endpoint",
 		Required: true,
 	}
+	getEndpointDescription := func(ctx *cli.Context) (string, error) {
+		description := ctx.String(endpointDescriptionOptionalFlag.Name)
+		descriptionFile := ctx.String(endpointDescriptionFileOptionalFlag.Name)
+		if description != "" && descriptionFile != "" {
```
Comment (mastermanu) [inline app/nexus.go:310]: we should also guard against somebody calling unset and also setting the description

Comment (mastermanu) [reply app/nexus.go:310]: ignore

**Review by mastermanu — APPROVED** [review]

**Pull request merged**